agent-connection/src/main/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/IndyParty.kt

```diff
@@ -86,5 +90,37 @@ class IndyParty(private val webSocket: AgentWebSocketClient, val did: String, va
      * @return observable (Single<>) object emitting a single [ProofInfo] upon subscription
      */
     override fun receiveProof() = webSocket.receiveClassObject<ProofInfo>(this)
+
+    /**
+     * Returns observable ([Single]<>) object, emitting JSON-encocoded Tails file by the given tails hash.
+     *
+     * @param tailsHash string-encocoded Tails file
+     *
+     * @return observable ([Single]<>) object emitting [TailsResponse] object
+     */
+    override fun requestTails(tailsHash: String) : Single<TailsResponse> {
+        webSocket.sendClassObject(TailsRequest(tailsHash), this)
```

Comment:
Shouldn`t we first sing for data, and after send request?

Reply:
fixed